Userland/Applets/Keymap/KeymapStatusWidget.cpp

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright (c) 2022, the SerenityOS developers.
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include "KeymapStatusWidget.h"
+#include "LibGUI/ActionGroup.h"
+#include <LibCore/Process.h>
+#include <LibGUI/Action.h>
+#include <LibGUI/ConnectionToWindowMangerServer.h>
+#include <LibGUI/Painter.h>
+#include <LibGfx/Point.h>
+#include <LibKeyboard/CharacterMap.h>
+
+void KeymapStatusWidget::mousedown_event(GUI::MouseEvent& event)
+{
+    Gfx::IntPoint point(event.x(), event.y());
+    MUST(refresh_menu());
+    m_context_menu->popup(point.translated(this->screen_relative_rect().location()));
+}
+
+ErrorOr<void> KeymapStatusWidget::refresh_menu()
+{
+    m_keymaps_group.for_each_action([&](auto& action) {
+        m_keymaps_group.remove_action(action);
+        return IterationDecision::Continue;
+    });
+
+    m_context_menu = GUI::Menu::construct();
+
+    auto mapper_config = TRY(Core::ConfigFile::open("/etc/Keyboard.ini"));
+    auto keymaps_string = mapper_config->read_entry("Mapping", "Keymaps", "");
+    auto keymaps = keymaps_string.split(',');
+
+    for (auto& keymap : keymaps) {
+        auto action = GUI::Action::create_checkable(keymap, [=](auto&) {
+            GUI::ConnectionToWindowMangerServer::the().async_set_keymap(keymap);
+        });
+
+        action->set_checked(keymap == m_current_keymap);
+
+        m_keymaps_group.add_action(action);
+        m_context_menu->add_action(action);
+    }
+
+    m_keymaps_group.set_exclusive(true);
+
+    m_context_menu->add_separator();
+
+    m_context_menu->add_action(GUI::Action::create("&Settings", [](auto&) {
```

Comment:
Could you add an icon? :)
```suggestion
    m_context_menu->add_action(GUI::Action::create("&Settings", Gfx::Bitmap::try_load_from_file("/res/icons/16x16/settings.png").release_value_but_fixme_should_propagate_errors(), [](auto&) {
```
![preview](https://user-images.githubusercontent.com/16520278/161605211-0587f9cd-08dd-4e23-bdf2-3538ad66fd9b.png)

Reply:
Nice idea!
Added